helm-charts/support/values.yaml

```diff
@@ -24,6 +29,19 @@ prometheus:
   pushgateway:
     enabled: false
   server:
+    ingress:
+      annotations:
+        # Annotations required to enable basic authentication for any ingress
+        # into prometheus server from the outside world - this secret is
+        # created via the prometheus-htpasswd.yaml file, controlled by config
```

Comment:
I don't understand that "this secret is created via the prometheus-htpasswd.yaml file, can you elaborate?

To clarify my confusion, I'm thinking `"The referenced k8s Secret is created by the Helm chart via config under prometheusIngressBasicAuth"`, and I lack knowledge about a `prometheus-htpasswd.yaml` file. I figure the k8s Secret will be consumed by the ingress controller independent of any prometheus logic.

Reply:
@consideRatio https://github.com/2i2c-org/infrastructure/pull/1091/files#diff-7042b93ab64b7c665e0172a141323f9e75f60d0fc7d06fce61dad6bc4ba41fff is prometheus-htpasswd.yaml, the actual file that is creating the k8s secret.

Reply:
Clarified by renaming file as well as saying `templates/<filename>`

Reply:
Oh when you said "the file that is creating the k8s secret" I got thinking something like `kubectl create secret generic --from-file ...`. But I understand it is a Helm chart template file that renders the k8s Secret resource! I figure if we clarify that it is the file mentioned is a helm template, it would be a lot clearer to me.

Can we also name the helm chart template file with a suffix of `secret.yaml`? It is very common practice in helm charts to name template files in a way that includes the k8s resource type, and it has been written in the Helm docs in the past at least that they suggest one resource per template file. Currently I see they have updated [their docs](https://helm.sh/docs/chart_template_guide/getting_started/) to say the less strict advice of:

> TIP: Template names do not follow a rigid naming pattern. However, we recommend using the suffix .yaml for YAML files and .tpl for helpers.

Reply:
@consideRatio ah, I'm not a big fan of adding the type of resource being created to the name of the file, preferring to signal intent instead. Just naming it the same as resource type works great when you're deploying a single application, or when it's under a subdirectory whose name explains what is being deployed. I'd like to not have both a name that explains intent and a suffix. It also reminds me a lot of [hungarian notation](https://en.wikipedia.org/wiki/Hungarian_notation). I'd prefer to keep it out if that's ok with you.

I've reworded the comment to make it clearer both where the secret is generated, as well as what feeds content in there.

Reply:
Oh okay - It gives me some itch as the convention is strong and has never been broken in the helm chart's I recall seeing besides the occational `rbac.yaml` file that includes multiple rbac related resource kinds - which in its name indicates what kind of resources it includes though.

Let's go with this though if you feel strongly about it!

Reply:
@consideRatio i've made a directory called prometheus-ingress-auth and moved this to a secret.yaml file there. I think this should both satisfy your itch as well as my desire to avoid hungarian notation.

Reply:
:D You are a kind person Yuvi!

Reply:
@consideRatio 
![image](https://user-images.githubusercontent.com/30430/158300045-96499df3-74be-404c-967b-1bded7daf943.png)